util/dasharo/tag-with-changelog

```diff
@@ -18,62 +18,62 @@ then
 	exit 1
 fi
 
-# Try to verify that the repo is clean before losing state.
-if ! git diff-index --quiet --cached HEAD 2>/dev/null || \
-	[ "$(git diff origin/main --shortstat 2>/dev/null | tail -n1)" != "" ]; then
-	echo "ERROR: repo is not clean.  Exiting." >&2
-	exit 1
-fi
-
 # Verify the command line arguments
 if [ "$1" == "--help" ] || [ -z "$1" ] || [ -z "$2" ]; then
 	echo
 	echo "Usage: $0 <old_version> <new_version> [release notes file]"
 	echo "Old version should be a tag (4.1), a branch (origin/4.1), or a commit id"
-	echo "New version can be a branch (origin/main) a tag (4.2), or a commit id"
-	echo "Logfile can be a new file or an existing file to update"
+	echo "New version must be a name of a new tag, this script fails if it exists"
```

Comment:
The script no longer fails if the new tag exists because that was a side-effect of creating the tag.  This text or the code needs to change.